lsp/server/src/utils/orgUtils.ts

```diff
@@ -0,0 +1,327 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import {
+    AuthInfo,
+    ConfigAggregator,
+    Connection,
+    OrgConfigProperties,
+    StateAggregator
+} from '@salesforce/core';
+import * as os from 'os';
+import * as fs from 'fs';
+import * as path from 'path';
+import { WorkspaceUtils } from './workspaceUtils';
+import { ObjectInfoRepresentation } from '../types';
+
+enum AuthStatus {
+    UNKNOWN,
+    AUTHORIZED,
+    UNAUTHORIZED
+}
+
+export class OrgUtils {
+    private static orgName: string = '';
+    private static objectInfoFolder = 'objectInfos';
+    private static entityListFileName = 'entity_list.json';
+    private static connection: Connection | undefined;
+
+    private static authStatus: AuthStatus = AuthStatus.UNKNOWN;
+
+    private static objectInfoInMemoCache = new Map<
+        string,
+        ObjectInfoRepresentation
+    >();
+    private static objectInfoPromises = new Map<
+        string,
+        Promise<ObjectInfoRepresentation | undefined>
+    >();
+    private static entities: string[] = [];
+
+    private static sfdxDirWatcher: fs.FSWatcher | undefined;
+    private static sfDirWatcher: fs.FSWatcher | undefined;
+
+    private static sfdxFolder = '.sfdx';
+    /**
+     * The global folder in which sf state is stored.
+     */
+    private static sfFolder = '.sf';
+
+    private static get SFDX_DIR() {
+        return path.join(os.homedir(), this.sfdxFolder);
+    }
+    /**
+     * The full system path to the global sf state folder.
+     */
+    private static get SF_DIR() {
+        return path.join(os.homedir(), this.sfFolder);
+    }
+
+    // Retrieves default organiztion's name.
+    private static async getDefaultOrg(): Promise<string> {
+        const aggregator = await ConfigAggregator.create();
+
+        await aggregator.reload();
+
+        const currentUserConfig = aggregator.getInfo(
+            OrgConfigProperties.TARGET_ORG
+        );
+
+        if (currentUserConfig.value) {
+            this.orgName = currentUserConfig.value.toString();
+            return Promise.resolve(this.orgName);
+        }
+        return Promise.reject('no org');
```

Comment:
You can throw error with `throw new Error('no org')`. This has benefit of passing stack trace to the caller.

Reply:
This is not an error if default org does not exist

Reply:
It's not useful because `reject` will go into caller's `catch`, in `getDefaultUserName()`.  There you are not using `'no org'` in any way. All you are doing is catch an error and return `undefined`. If you throw error like this

`throw new Error('no org')` that will still land in the `catch` block of the same function. And then in there you can access `'no org'` in the form of `error.message` should you decide to log it.